api/tools/src/main/java/org/pmiops/workbench/tools/DeleteWorkspaces.java

```diff
@@ -0,0 +1,175 @@
+package org.pmiops.workbench.tools;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.Clock;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.dao.UserRecentWorkspaceDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbStorageEnums;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.firecloud.ApiClient;
+import org.pmiops.workbench.firecloud.FireCloudConfig;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.firecloud.FireCloudServiceImpl;
+import org.pmiops.workbench.firecloud.api.ProfileApi;
+import org.pmiops.workbench.firecloud.api.WorkspacesApi;
+import org.pmiops.workbench.model.WorkspaceActiveStatus;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.pmiops.workbench.workspaces.WorkspaceServiceImpl;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Scope;
+
+@Configuration
+@Import({FireCloudServiceImpl.class, FireCloudConfig.class})
+public class DeleteWorkspaces {
+
+  private static final Logger log = Logger.getLogger(DeleteWorkspaces.class.getName());
+
+  private static Option deleteListFilename =
+      Option.builder()
+          .longOpt("delete-list-filename")
+          .desc(
+              "File containing list of workspaces to delete. Each line should contain a single workspace's namespace and firecloud name, separated by a comma"
+                  + "Example: ws-namespace-1,fc-id-1 \n ws-namespace-2,fc-id-2 \n ws-namespace-3, fc-id-3")
+          .required()
+          .hasArg()
+          .build();
+
+  private static Options options = new Options().addOption(deleteListFilename);
+
+  @Bean
+  public WorkspaceService workspaceService(
+      FireCloudService fireCloudService,
+      Clock clock,
+      WorkspaceDao workspaceDao,
+      UserRecentWorkspaceDao userRecentWorkspaceDao,
+      Provider<DbUser> dbUserProvider) {
+    return new WorkspaceServiceImpl(
```

Comment:
future: if we determine a useful core of the workspaceService we could break that into its own service and inject the whole thing.